tests/std/context_timeout_test.go

```diff
@@ -40,34 +40,27 @@ func TestStdContextStdTimeout(t *testing.T) {
 		t.Run(fmt.Sprintf("%s Protocol", name), func(t *testing.T) {
 			connect, err := GetStdDSNConnection(protocol, useSSL, nil)
 			require.NoError(t, err)
-			{
+
+			t.Run("query which triggers timeout", func(t *testing.T) {
 				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 				defer cancel()
 				if row := connect.QueryRowContext(ctx, "SELECT 1, sleep(3)"); assert.NotNil(t, row) {
 					var a, b int
 					if err := row.Scan(&a, &b); assert.Error(t, err) {
-						switch err := err.(type) {
-						case *net.OpError:
-							assert.Equal(t, "read", err.Op)
-						case *url.Error:
-							assert.Equal(t, context.DeadlineExceeded, err.Err)
-						default:
-							assert.ErrorIs(t, err, context.DeadlineExceeded)
-						}
+						chtesting.AssertIsTimeoutError(t, err)
 					}
 				}
-			}
-			{
-				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+			})
+			t.Run("query which returns in time", func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Note for the reviewer:

This particular context driven deadline is now being checked on each read down in the net stack (it is not being removed after first read) this test became unstable. I tweaked the timeout running the tests with `-count 10` until I could get something consistent. Three seconds seemed enough to process all of the response.